### ISSUE TITLE
feat: learned-risk trainer entrypoint + config slice (unblocks draft PR #4552, part of #1472)

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#4617]
Prerequisite slice unblocking draft PR #4552 (learned-risk Slurm execution contract), per the gate verdict and parking decision recorded there.

## Task
Implement the real trainer entrypoint + config that #4552's `slurm_execution` block cites as must-exist paths:
1. `scripts/training/train_learned_risk_model.py` — entrypoint consistent with the #1472 launch-packet contract (reads the packet config, materialized trace manifest inputs, writes the expected output root / status artifact); CPU-smoke runnable (tiny synthetic input), NO Slurm submission.
2. `configs/training/learned_risk_model_v1.yaml` — config consistent with `configs/training/learned_risk_model_issue_1395_launch_packet.yaml` conventions.
3. Focused tests: entrypoint importable + arg contract; smoke run on synthetic data produces the status artifact.

## After merge
Rebase PR #4552 so its cited paths resolve, confirm docs-evidence-integrity green, then `gh pr ready 4552` hands it back to the gate. (That step is recorded on #4552 itself.)

## Boundary
Read the FULL #1472 thread first. No Slurm submission, no campaign claims. If the trace-manifest inputs the trainer needs are genuinely undefined (see #4586), post ONE blocked-comment naming the gap and exit clean.

**Comment by ll7** [comment]
## Implementation plan for #4617

### Goal

Add the real learned-risk trainer entrypoint and training config paths that draft PR #4552 cites in its `slurm_execution` contract, while keeping the slice CPU-smoke-only and fail-closed on unresolved trace-manifest inputs.

No Slurm submission, no full training campaign, and no learned-risk claim promotion.

### Current context

Primary issue: #1472. Draft PR #4552 adds launch-packet validation for a `slurm_execution` block and remains draft because cited paths do not yet exist.

Expected new paths:

```text
scripts/training/train_learned_risk_model.py
configs/training/learned_risk_model_v1.yaml
```

Related readiness blocker: #4586 / trace manifest still has pending aliases and retrieval status blockers. The trainer must be able to run a synthetic CPU smoke, but real training must remain gated on the trace manifest.

### Implementation steps

1. **Read and mirror the #1472 launch-packet contract**

   Ground implementation against:

   ```text
   configs/training/learned_risk_model_issue_1395_launch_packet.yaml
   configs/training/learned_risk_trace_manifest_issue_2312.yaml
   scripts/validation/validate_learned_risk_launch_packet.py
   scripts/validation/check_learned_risk_campaign_readiness.py
   robot_sf/training/learned_risk_campaign_readiness.py
   ```

   Required contract elements:

   - candidate id: `learned_risk_model_v1`;
   - labels: `collision`, `near_miss`, `low_progress`;
   - required fields: `scenario_id`, `seed`, `candidate_id`, `termination_reason`, `metrics`, `trajectory_features`, `labels`;
   - hard guards remain authoritative;
   - learned risk is auxiliary-cost-only;
   - real traces must come from durable manifest inputs, not ad-hoc `output/` paths.

2. **Add training config**

   Add:

   ```text
   configs/training/learned_risk_model_v1.yaml
   ```

   Suggested fields:

   ```yaml
   schema_version: learned-risk-model-training.v1
   candidate_id: learned_risk_model_v1
   launch_packet: configs/training/learned_risk_model_issue_1395_launch_packet.yaml
   trace_manifest: configs/training/learned_risk_trace_manifest_issue_2312.yaml
   output_root: artifacts/training/learned_risk_model_v1
   status_artifact_path: artifacts/training/learned_risk_model_v1/status.json
   labels: [collision, near_miss, low_progress]
   features:
     - candidate_command_count
     - min_rollout_clearance_m
     - mean_pedestrian_distance_m
     - route_progress_delta
     - metrics.min_distance
     - metrics.avg_speed
     - metrics.goal_progress
   model:
     type: lightweight_logistic_or_mlp
     hidden_dims: [32, 16]
   training:
     max_epochs: 5
     batch_size: 64
     seed: 1472
   diagnostics:
     required: [auroc, auprc, brier, false_negative_rate, scenario_stratified_errors]
   claim_boundary: launch_entrypoint_and_cpu_smoke_only_no_slurm_no_claim
   ```

   Keep output/log/status paths aligned with #4552’s launch-packet expectations. If #4552 names different exact paths, match those exactly.

3. **Add trainer entrypoint**

   Add:

   ```text
   scripts/training/train_learned_risk_model.py
   ```

   CLI requirements:

   ```bash
   uv run python scripts/training/train_learned_risk_model.py \
     --config configs/training/learned_risk_model_v1.yaml \
     --status-out <path> \
     --output-root <path> \
     --smoke
   ```

   Required behavior:

   - loads training config;
   - validates launch packet;
   - checks trace-manifest readiness for real mode;
   - in `--smoke` mode, uses a tiny synthetic fixture trace and does not require private artifacts;
   - in real mode, refuses to run unless `check_learned_risk_campaign_readiness.py` would be launch-ready;
   - writes a status artifact in a stable JSON shape;
   - writes explicit claim boundary and `slurm_submission: false`.

4. **Implement minimal model/training logic**

   Keep it simple and testable. Acceptable first slice:

   - parse rows into feature matrix and per-label binary targets;
   - train one lightweight sklearn-like or PyTorch model per label, or a simple multi-head MLP if dependencies are already present;
   - compute smoke diagnostics on the synthetic fixture;
   - in real mode, leave full training to Slurm once manifest inputs are ready.

   If adding full model code is too large, implement a deterministic smoke trainer that exercises parsing, label validation, metrics, and status artifact shape, and fail closed for real mode until durable traces are ready. The issue asks for an entrypoint/config slice, not a full Slurm run.

5. **Status artifact contract**

   Status JSON should include:

   - schema version;
   - candidate id;
   - launch packet path;
   - trace manifest path;
   - mode: `smoke` or `real`;
   - training state: `smoke_completed`, `blocked_trace_manifest`, `training_completed`, or `failed`;
   - labels and feature list;
   - row counts;
   - diagnostics available;
   - output paths;
   - claim boundary;
   - hard-guard auxiliary-role statement.

6. **Tests**

   Add tests, for example:

   ```text
   tests/training/test_train_learned_risk_model.py
   ```

   Required coverage:

   - entrypoint imports and `--help` works;
   - config loads and matches launch-packet candidate/labels;
   - synthetic smoke run writes status artifact;
   - smoke status contains candidate id, labels, diagnostics, and claim boundary;
   - real mode blocks when trace manifest is not ready;
   - missing required trace fields fail closed with a clear error;
   - output/status paths are not worktree-local `output/` if the launch-packet contract forbids them.

7. **Rebase/ready handoff for PR #4552 after merge**

   After this issue merges:

   - rebase #4552 onto updated main;
   - confirm cited `slurm_execution.entrypoint` and config paths exist;
   - run docs-evidence-integrity / launch-packet validation;
   - mark #4552 ready for review.

   Do not do that inside this PR unless explicitly authorized; it is recorded on #4552.

### Verification commands

```bash
uv run ruff check scripts/training/train_learned_risk_model.py tests/training/test_train_learned_risk_model.py
uv run pytest tests/training/test_train_learned_risk_model.py -q
uv run python scripts/training/train_learned_risk_model.py --help
uv run python scripts/training/train_learned_risk_model.py \
  --config configs/training/learned_risk_model_v1.yaml \
  --smoke \
  --output-root output/issue_4617_learned_risk_smoke \
  --status-out output/issue_4617_learned_risk_smoke/status.json
uv run python scripts/validation/validate_learned_risk_launch_packet.py \
  --config configs/training/learned_risk_model_issue_1395_launch_packet.yaml --json
uv run python scripts/validation/check_learned_risk_campaign_readiness.py --json
```

The readiness command may still exit blocked because #4586 is unresolved; that is acceptable if the trainer real mode also blocks cleanly.

### Acceptance criteria

- `scripts/training/train_learned_risk_model.py` exists and is importable/CLI-runnable.
- `configs/training/learned_risk_model_v1.yaml` exists and is consistent with #1472 launch-packet labels/features.
- Synthetic CPU smoke writes the expected status artifact.
- Real mode fails closed while trace-manifest inputs remain unresolved.
- No Slurm submission, full training run, checkpoint publication, or claim promotion occurs.
